src/h5web/visualizer/Visualizer.test.tsx

```diff
@@ -1,32 +1,138 @@
-    expect(scalarTab).toHaveAttribute('aria-selected', 'true');
-    expect(screen.getByText('0')).toBeVisible();
+  test('show "Scalar" tab for scalar dataset with basic type', async () => {
+    renderApp();
 
-    fireEvent.click(screen.getByRole('treeitem', { name: 'scalar_str' }));
-    expect(await screen.findByText('foo')).toBeVisible();
+    fireEvent.click(await screen.findByRole('treeitem', { name: 'entities' }));
+    fireEvent.click(screen.getByRole('treeitem', { name: 'scalar_int' }));
+
+    const tabs = await findVisSelectorTabs();
+    expect(tabs).toHaveLength(1);
+    expect(tabs[0]).toHaveTextContent(Vis.Scalar);
+  });
+
+  test('show "Matrix" and "Line" tabs for 1D dataset', async () => {
+    renderApp();
+
+    fireEvent.click(
+      await screen.findByRole('treeitem', { name: 'nD_datasets' })
+    );
+    fireEvent.click(screen.getByRole('treeitem', { name: 'oneD' }));
+
+    const tabs = await findVisSelectorTabs();
+    expect(tabs).toHaveLength(2);
+    expect(tabs[0]).toHaveTextContent(Vis.Matrix);
+    expect(tabs[1]).toHaveTextContent(Vis.Line);
+    expect(tabs[1]).toHaveAttribute('aria-selected', 'true');
+  });
+
+  test('show "Matrix", "Line" and "Heatmap" tabs for 2D datasets', async () => {
+    renderApp();
+
+    fireEvent.click(
+      await screen.findByRole('treeitem', { name: 'nD_datasets' })
+    );
+    fireEvent.click(screen.getByRole('treeitem', { name: 'twoD' }));
+
+    const tabs = await findVisSelectorTabs();
+    expect(tabs).toHaveLength(3);
+    expect(tabs[0]).toHaveTextContent(Vis.Matrix);
+    expect(tabs[1]).toHaveTextContent(Vis.Line);
+    expect(tabs[2]).toHaveTextContent(Vis.Heatmap);
+    expect(tabs[2]).toHaveAttribute('aria-selected', 'true');
+  });
+
+  test('show "NX Spectrum" tab for NXdata group with "spectrum" interpretation', async () => {
+    renderApp();
+
+    fireEvent.click(
+      await screen.findByRole('treeitem', { name: 'nexus_entry' })
+    );
+    fireEvent.click(screen.getByRole('treeitem', { name: 'spectrum' }));
+
+    const tabs = await findVisSelectorTabs();
+    expect(tabs).toHaveLength(1);
+    expect(tabs[0]).toHaveTextContent(Vis.NxSpectrum);
   });
 
-  test('switch visualisations', async () => {
+  test('show "NX Image" tab for NXdata group with "image" interpretation', async () => {
+    renderApp();
+
+    fireEvent.click(
+      await screen.findByRole('treeitem', { name: 'nexus_entry' })
+    );
+    fireEvent.click(screen.getByRole('treeitem', { name: 'image' }));
+
+    const tabs = await findVisSelectorTabs();
+    expect(tabs).toHaveLength(1);
+    expect(tabs[0]).toHaveTextContent(Vis.NxImage);
+  });
+
+  test('show "NX Image" tab for NXdata group with 2D signal', async () => {
+    renderApp();
+
+    fireEvent.click(
+      await screen.findByRole('treeitem', { name: 'nexus_entry' })
+    );
+    fireEvent.click(screen.getByRole('treeitem', { name: 'nx_process' }));
+    fireEvent.click(screen.getByRole('treeitem', { name: 'nx_data' }));
```

Comment:
Didn't we talk about adding a helper to select nested entities more easily ?

Reply:
Yeah, I was planning on doing this soon.